internal/snapshotstrategy/strategy.go

```diff
@@ -1,12 +1,12 @@
 // SPDX-FileCopyrightText: 2023 SAP SE or an SAP affiliate company and IronCore contributors
 // SPDX-License-Identifier: Apache-2.0
 
-package utils
+package snapshotstrategy
```

Comment:
I think it's not 100% correct, to have both strategies in the same package. Should we maybe rename it to `strategy`?